application/helpers/SortHelper.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+* This file is part of LimeSurvey
+* Offer some helper for sorting
+*/
+
+namespace LimeSurvey\Helpers;
+
+use Yii;
+
+// Can not use Collator class, must work without
+
+class SortHelper
+{
+    /** @var integer flag for regular sort */
+    const SORT_REGULAR = 0;
+    /** @var integer flag for numeric sort */
+    const SORT_NUMERIC = 1;
+    /** @var integer flag for string sort */
+    const SORT_STRING = 2;
+
+    /**
+     * @var self Singleton
+     */
+    private static $instance = null;
+
+    /**
+     * @var string current language
+     */
+    private static $language = null;
+    /**
+     * @var \Collator if possible
+     */
+    private static $collator = null;
+
+    /**
+     * Set the language when construct
+     */
+    private function __construct($language)
+    {
+        self::$language = $language;
+        if (class_exists('Collator')) {
+            $locale = $language;
+            $languageData = getLanguageData();
+            if (isset($languageData[$language]['cldr'])) {
+                $locale = $languageData[$language]['cldr'];
+            }
+            self::$collator = new \Collator($locale);
```

Comment:
Does it wort caching the collator as in the cases this is called inside a for-loop? Not sure if itis a heavy load thing